engine/src/main/java/org/terasology/engine/rendering/dag/AbstractNode.java

```diff
@@ -197,7 +197,7 @@ public boolean addOutputBufferPairConnection(int id, BufferPair bufferPair) {
             if (!localBufferPairConnection.getConnectedConnections().isEmpty()) {
                 logger.debug("Propagating bufferPair data to all connected connections of {}: ", localBufferPairConnection);
                 localBufferPairConnection.getConnectedConnections().forEach((k, v) -> {
-                    logger.debug("setting data for: {} ,", v);
+                    logger.debug("setting data for: {} ,", v.toString());
```

Comment:
Why do we need explicit `toString()` here?

As far as I understood, `toString()` is called implicitly for any non-string objects.
Also, explicit `toString()` might induce additional cost even if debug logging is disabled while the implicit call is only made if the log is actually being written (debug logging enabled).
(see https://stackoverflow.com/questions/74123094/cost-of-tostring-in-logging or https://stackoverflow.com/questions/50160672/tostring-in-logback; kudos @skaldarnar for the references)

Reply:
to be honest, did not check at all. it was there before, intellij showed in many cases to remove toString, and here not. ill drop the pull request.